cmd/kube-proxy/app/server.go

```diff
@@ -263,6 +271,39 @@ func (o *Options) Complete() error {
 	return utilfeature.DefaultMutableFeatureGate.SetFromMap(o.config.FeatureGates)
 }
 
+// copyLogsFromFlags applies the logging flags from the given flag set to the given
+// configuration. Fields for which the corresponding flag was not used are left
+// unmodified. For fields that have multiple values (like vmodule), the values from
+// the flags get joined so that the command line flags have priority.
+//
+// TODO (pohly): move this to logsapi
```

Comment:
Moving this after the 1.28 released?

Reply:
yes, it has to be later

Reply:
Let's to it in 1.29.

Reply:
Can it be moved now? I can work on this.

Reply:
Yes, and please do.

Some points to consider:
- Can the same be done for apps using just plain Go flags? The "flag was set" field only exists for pflag. A potential solution might be to wrap the values in a flag.Value implementation which tracks whether `Set` was called.
- Examples and docs...
  - The original plan was to have a JSON or YAML config that gets decoded into the LoggingConfiguration instance after parsing, in which case this new function isn't needed. It needs to be checked that this works and be documented.
  - For configs where that doesn't work (i.e. anything based on registered types via apimachinery like this  config here), the new functions need to be documented.

Reply:
> Can the same be done for apps using just plain Go flags?

Is there any difference between Go Flags and Pflags? I guess this needs to be tested.

> Examples and docs

Where should they be placed?

Reply:
pflag.Flag is a larger struct than flag.Flag.

Examples can go into https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/component-base/logs/example

Reply:
Thanks!

Reply:
> A potential solution might be to wrap the values in a flag.Value implementation which tracks whether Set was called.

I found it hard to check if a go flag was set,  I tried to check it by using the following codes, but it didn't work.

```go
func isFlagSet(name string) bool {
    found := false
    flag.Visit(func(f *flag.Flag) {
        if f.Name == name {
            found = true
        }
    })
    return found
}
```

Could you give me an example?

Reply:
and we may need a new function called `copyLogsFromGoFlags ` for this, It may not be for Kubernetes users.

Reply:
A wrapper as in this example is one possible solution:
```Go
package main

import (
	"flag"
	"fmt"
)

type flagWrapper struct {
	flag.Value
	isSet bool
}

func (f *flagWrapper) Set(value string) error {
	f.isSet = true
	return f.Value.Set(value)
}

var _ flag.Value = &flagWrapper{}

func main() {
	var fs flag.FlagSet
	fs.Int("some-int", 42, "the answer")
	arg := fs.Lookup("some-int")
	wrapper := &flagWrapper{Value: arg.Value}
	flag.CommandLine.Var(wrapper, arg.Name, arg.Usage)
	flag.Parse()
	fmt.Printf("some-int set = %v\n", wrapper.isSet)
}
```

The downside is that `-h` doesn't know about the actual type of the parameter:
```console
$ go run . -h
Usage of /tmp/go-build544671316/b001/exe/testflag:
  -some-int value
    	the answer
```

Without the wrapper, `int` would be shown instead of `value`. That's because `flag` looks at the type of a value and has special support for builtin value types.

Reply:
Thanks for the example. 

> The downside is that -h doesn't know about the actual type of the parameter

Can we explain in the usage what the actual type of the parameter is?

Reply:
Yes. Showing "value" isn't a showstopper.